consensus/src/main/kotlin/maru/consensus/ValidatorProvider.kt

```diff
@@ -19,6 +19,27 @@ import maru.core.BeaconBlockHeader
 import maru.core.Validator
 import tech.pegasys.teku.infrastructure.async.SafeFuture
 
+/**
+ * Provides access to the set of validators for a given block.
+ */
 interface ValidatorProvider {
   fun getValidatorsForBlock(header: BeaconBlockHeader): SafeFuture<Set<Validator>>
```

Comment:
If there are methods to get validators by block number, why would user use method that accepts the header if they can just pass block number? I think we should remove this method now

Reply:
I saw Gaurav's comment now. I guess it will be removed later